sdk-extension/opentelemetry-sdk-extension-aws/tests/trace/propagation/test_aws_xray_format.py

```diff
@@ -0,0 +1,375 @@
+# limitations under the License.
+
+import unittest
+
+from requests.structures import CaseInsensitiveDict
+
+import opentelemetry.trace as trace_api
+from opentelemetry.sdk.extension.aws.trace.propagation.aws_xray_format import (
+    AwsXRayFormat,
+)
+from opentelemetry.trace import (
+    DEFAULT_TRACE_OPTIONS,
+    DEFAULT_TRACE_STATE,
+    INVALID_SPAN_CONTEXT,
+    SpanContext,
+    TraceFlags,
+    TraceState,
+    set_span_in_context,
+)
+
+TRACE_ID_BASE16 = "8a3c60f7d188f8fa79d48a391a778fa6"
+
+SPAN_ID_BASE16 = "53995c3f42cd8ad8"
+
+# Propagators Usage Methods
+
+
+def get_as_list(dict_object, key):
+    value = dict_object.get(key)
+    return [value] if value is not None else []
+
+
+# Inject Methods
+
+
+def build_test_context(
+    trace_id=int(TRACE_ID_BASE16, 16),
+    span_id=int(SPAN_ID_BASE16, 16),
+    is_remote=True,
+    trace_flags=DEFAULT_TRACE_OPTIONS,
+    trace_state=DEFAULT_TRACE_STATE,
+):
+    return set_span_in_context(
+        trace_api.DefaultSpan(
+            SpanContext(
+                trace_id, span_id, is_remote, trace_flags, trace_state,
+            )
+        )
+    )
+
+
+def build_dict_with_xray_trace_header(
+    trace_id="{}{}{}{}{}".format(
+        AwsXRayFormat.TRACE_ID_VERSION,
+        AwsXRayFormat.TRACE_ID_DELIMITER,
+        TRACE_ID_BASE16[: AwsXRayFormat.TRACE_ID_FIRST_PART_LENGTH],
+        AwsXRayFormat.TRACE_ID_DELIMITER,
+        TRACE_ID_BASE16[AwsXRayFormat.TRACE_ID_FIRST_PART_LENGTH :],
+    ),
+    span_id=SPAN_ID_BASE16,
+    sampled="0",
+):
+    carrier = CaseInsensitiveDict()
+
+    carrier[AwsXRayFormat.TRACE_HEADER_KEY] = "".join(
+        [
+            "{}{}{}{}".format(
+                AwsXRayFormat.TRACE_ID_KEY,
+                AwsXRayFormat.KEY_AND_VALUE_DELIMITER,
+                trace_id,
+                AwsXRayFormat.KV_PAIR_DELIMITER,
+            ),
+            "{}{}{}{}".format(
+                AwsXRayFormat.PARENT_ID_KEY,
+                AwsXRayFormat.KEY_AND_VALUE_DELIMITER,
+                span_id,
+                AwsXRayFormat.KV_PAIR_DELIMITER,
+            ),
+            "{}{}{}".format(
+                AwsXRayFormat.SAMPLED_FLAG_KEY,
+                AwsXRayFormat.KEY_AND_VALUE_DELIMITER,
+                sampled,
+            ),
+        ]
+    )
+
+    return carrier
+
+
+# Extract Methods
+
+
+def get_extracted_span_context(encompassing_context):
+    return trace_api.get_current_span(encompassing_context).get_span_context()
+
+
+class AwsXRayPropagatorTest(unittest.TestCase):
+    carrier_setter = CaseInsensitiveDict.__setitem__
+    carrier_getter = get_as_list
+    XRAY_PROPAGATOR = AwsXRayFormat()
+
+    # Inject Tests
+
+    def test_inject_into_non_sampled_context(self):
+        carrier = CaseInsensitiveDict()
+
+        AwsXRayPropagatorTest.XRAY_PROPAGATOR.inject(
+            AwsXRayPropagatorTest.carrier_setter, carrier, build_test_context()
+        )
+
+        self.assertTrue(
+            set(carrier.items()).issubset(
```

Comment:
Similarly, if the assertion has the string form of the header, it's much easier to see what it's actually comparing again. This is actually getting to the point where the unit test needs is own tests :P

Reply:
Sounds good, I split them out into `injected_items` and `expected_items` and used the raw string to hopefully make it easier to understand :)